File: 1019-squares-of-a-sorted-array/squares-of-a-sorted-array.java
class Solution {
    public int[] sortedSquares(int[] nums) {
        int size = nums.length - 1;
        int result[] = new int[size + 1]; 
        int left = 0;
        int right = size;
        int postion = result.length - 1;
        while(left <= right){
            int sqrLeft = nums[left] * nums[left];
            int sqrRight = nums[right] * nums[right];
            if(sqrLeft >= sqrRight){
                result[postion] = sqrLeft;
                left++;
            }else{
                result[postion] = sqrRight;
                right--;
            }
            postion--;
        }
        return result;
    }
}

File: 1019-squares-of-a-sorted-array/squares-of-a-sorted-array.ts
function sortedSquares(nums: number[]): number[] {
    let size:number = nums.length - 1;
    let result:number[] = new Array(size + 1).fill(0);
    let left:number = 0;
    let right:number = size;
    let position:number = result.length - 1;

    while(left <= right){
        let squaredLeft:number = nums[left] ** 2;
        let squaredRight:number = nums[right] ** 2;

        if(squaredLeft <= squaredRight){
            result[position] = squaredRight;
            right--;
        }else{
            result[position] = squaredLeft;
            left++;
        }
        position--;
    }
    return result;


};